Not able to use node-config library with turbopack

### Link to the code that reproduces this issue

https://github.com/sdoomz/nextjs-turbopack-node-config-bug

### To Reproduce

1. clone repo https://github.com/sdoomz/nextjs-turbopack-node-config-bug
2. npm install 
3. npm run dev


### Current vs. Expected behavior

**Current behavior:**

Getting following error when using [node-config](https://github.com/node-config/node-config) in server code (app/page.tsx): 

```
Module not found: Can't resolve 'coffee-script'
```

**Expected behavior:**
No error thrown

### Provide environment information

```bash
Operating System:
  Platform: darwin
  Arch: arm64
  Version: Darwin Kernel Version 24.3.0: Thu Jan  2 20:24:22 PST 2025; root:xnu-11215.81.4~3/RELEASE_ARM64_T6041
  Available memory (MB): 49152
  Available CPU cores: 14
Binaries:
  Node: 23.11.0
  npm: 10.9.2
  Yarn: N/A
  pnpm: N/A
Relevant Packages:
  next: 15.3.1 // Latest available version is detected (15.3.1).
  eslint-config-next: N/A
  react: 19.1.0
  react-dom: 19.1.0
  typescript: 5.8.3
Next.js Config:
  output: N/A
```

### Which area(s) are affected? (Select all that apply)

Turbopack

### Which stage(s) are affected? (Select all that apply)

next dev (local)

### Additional context

I tested my application with and without --turbopack flag and was able to reproduce this issue only in turbo mode.

I think It's quite important to support [node-config](https://github.com/node-config/node-config) since it's the most popular library for config management in nodejs. 

Error screenshot: 
<img width="1000" alt="Image" src="https://github.com/user-attachments/assets/d8bc1823-98f8-4512-9235-95f483da396c" />

--- a/docs/01-app/05-api-reference/05-config/01-next-config-js/serverExternalPackages.mdx
+++ b/docs/01-app/05-api-reference/05-config/01-next-config-js/serverExternalPackages.mdx
@@ -43,6 +43,7 @@ Next.js includes a [short list of popular packages](https://github.com/vercel/ne
 - `better-sqlite3`
 - `canvas`
 - `chromadb-default-embed`
+- `config`
 - `cpu-features`
 - `cypress`
 - `dd-trace`

--- a/docs/02-pages/04-api-reference/04-config/01-next-config-js/serverExternalPackages.mdx
+++ b/docs/02-pages/04-api-reference/04-config/01-next-config-js/serverExternalPackages.mdx
@@ -43,6 +43,7 @@ Next.js includes a [short list of popular packages](https://github.com/vercel/ne
 - `better-sqlite3`
 - `canvas`
 - `chromadb-default-embed`
+- `config`
 - `cpu-features`
 - `cypress`
 - `dd-trace`

--- a/packages/next/src/lib/server-external-packages.json
+++ b/packages/next/src/lib/server-external-packages.json
@@ -24,6 +24,7 @@
   "better-sqlite3",
   "canvas",
   "chromadb-default-embed",
+  "config",
   "cpu-features",
   "cypress",
   "dd-trace",
@@ -67,6 +68,5 @@
   "vscode-oniguruma",
   "webpack",
   "websocket",
-  "zeromq",
-  "config"
+  "zeromq"
 ]

--- a/packages/next/src/lib/server-external-packages.json
+++ b/packages/next/src/lib/server-external-packages.json
@@ -67,5 +67,6 @@
   "vscode-oniguruma",
   "webpack",
   "websocket",
-  "zeromq"
+  "zeromq",
+  "config"
 ]

